### ISSUE TITLE
JRoute::_() returns wrong url when the modern router is enabled and ID removing is on

### ISSUE THREAD

**Issue by philip-sorokin** [joomla/joomla-cms#14308]
### Steps to reproduce the issue

Download and install Joomla! 3.7 beta 3 with "Learning Joomla!" demo pages.
Enable modern routing with removing ID option.
Execute somewhere, e.g. in template view, the following code:
```
$route = ContentHelperRoute::getArticleRoute(8);
$sef = JRoute::_($route);

echo $sef;
```
### Expected result

It must return the following URL: 
`/using-joomla/extensions/components/content-component/article-category-list/beginners.html`

### Actual result

Instead, it returns a URL with ID: 
`/using-joomla/extensions/components/content-component/article-category-list/8-beginners.html`

### Additional comments

Sorry if I misunderstood the new routing system, but for me it looks strange.

**Comment by zero-24** [comment]
Where did you enable that option? com_content? Can you try to change that code in com_content?

**Comment by philip-sorokin** [comment]
Is there another place, where I can enable this option?

![routing](https://cloud.githubusercontent.com/assets/11632367/23522745/3ebb0d44-ff95-11e6-9cc4-d42313b51ef4.jpg)

**Comment by zero-24** [comment]
Yes you can do this per component but you need to enable this for e.g. com_newsfeeds again etc. and there is no general switch just a per component switch ;)

For com_content this is the right place to enable this :)

**Comment by Hackwar** [comment]
This should be fixed in #14322. Please test.
